### PR TITLE
[ios, macos] Sync offlineStorage singleton tests.

### PR DESCRIPTION
**Pull request by fabian-guerra**
This PR synchronizes the singleton access to avoid unit test interference.

### CODE REVIEW

File: platform/darwin/test/MGLOfflineStorageTests.mm [hunk 1]
```diff
@@ -33,32 +33,41 @@ + (void)tearDown {
     XCTAssertFalse([[NSFileManager defaultManager] fileExistsAtPath:cacheURL.path], @"Cache subdirectory should not exist.");
 }
 
+- (MGLOfflineStorage *)syncedOfflineStorage {
+    MGLOfflineStorage *os = nil;
+    @synchronized (self) {
```
Comment (julianrex) [inline platform/darwin/test/MGLOfflineStorageTests.mm]: I think this should be `@synchronized ([MGLOfflineStorage class])`, given that a new instance of `MGLOfflineStorageTests` is created for each test.

Comment (fabian-guerra) [reply platform/darwin/test/MGLOfflineStorageTests.mm]: Right but each test uses a different path so there's no need to sync the class.

Comment (julianrex) [reply platform/darwin/test/MGLOfflineStorageTests.mm]: Then the question is what is the `@synchronized (self)` for, given that `self` will be different for each test?

**Comment by friedbunny** [comment]
Instability in our offline tests has abated since #13959 — do we still want to implement these changes?

**Comment by julianrex** [comment]
> Instability in our offline tests has abated since #13943 — do we still want to implement these changes?

I think we should close this PR in this case. Ideally we wouldn't have a singleton for these tests, and that would bypass the need for this.

**Pull request closed without merging**